src/connector.rs

```diff
@@ -312,8 +312,11 @@ impl EthConnectorContract {
         // Check promise results
         let data0: Vec<u8> = match sdk::promise_result(0) {
             PromiseResult::Successful(x) => x,
-            _ => sdk::panic_utf8(b"ERR_PROMISE_INDEX"),
+            PromiseResult::Failed => sdk::panic_utf8(b"ERR_???"),
+            // This shouldn't be reachable
+            PromiseResult::NotReady => sdk::panic_utf8(b"ERR_PROMISE_NOT_READY"),
```

Comment:
Shouldn't we use `unreachable!` instead?

Reply:
It is better to fail with some significant error in case something doesn't work as we expect. In this case, we can reach this branch if something was missunderstood by us, or third party violates some invariant. I'd keep it as it is right now, rather than converting into unreachable!()

Reply:
I'm in the other camp if we are for sure that it can't be reachable. Of course, future logic could state otherwise. Perhaps, set the error message as unreachable to give a hint that this is basically unexpected?